### PR TITLE
Added disutils compilation options to CythonCodeWrapper class and autowrap.

### PR DESCRIPTION
**Pull request by moorepants**

### CODE REVIEW

File: sympy/utilities/autowrap.py [hunk 1]
```diff
@@ -245,10 +250,48 @@ class CythonCodeWrapper(CodeWrapper):
         "{declarations}"
         "{body}")
 
+    std_compile_flag = '-std=c99'
+
     def __init__(self, *args, **kwargs):
-        super(CythonCodeWrapper, self).__init__(*args, **kwargs)
+        """
+        The following optional parameters get passed to ``distutils.Extension``
+        for buidling the Python extension module. Read it's documentation to
```
Comment (asmeurer) [inline sympy/utilities/autowrap.py]: its

File: sympy/utilities/tests/test_autowrap.py [hunk 2]
```diff
@@ -87,6 +87,111 @@ def test_cython_wrapper_inoutarg():
...
+                             extra_compile_args=['-slow-math', '-std=c99'],
+                             extra_link_args=['-lswamp', '-ltrident']
+                            )],
+    )\
+"""
+
+    try:
+        code_gen._need_numpy = True
+        code_gen._prepare_files(routine)
+        with open('setup.py') as f:
```
Comment (asmeurer) [inline sympy/utilities/tests/test_autowrap.py]: This needs to be done in a temporary directory.

Comment (moorepants) [inline sympy/utilities/tests/test_autowrap.py]: I'd have to modify the `_prepare_files` method to do that. I could give an optional arg for a tempdir there, I suppose.

Comment (asmeurer) [inline sympy/utilities/tests/test_autowrap.py]: I don't know if that's the right place to do it, unless you are also going to remove the files in _prepare_files.

**Comment by moorepants** [comment]
Ok, I think this is ready. We'd like to get this and #12833 in for 1.1.

**Comment by bjodah** [comment]
I opened this (as a review):
https://github.com/moorepants/sympy/pull/6
against @moorepants branch.

**Comment by moorepants** [comment]
Seems like our travis builds are taking forever. This one didn't start for 20 hours. Guess we just have to wait. It's odd the travis status doesn't show any major issues.

**Comment by moorepants** [comment]
I'm considering this reviewed by @bjodah and I. I addressed Aaron's comments. All tests pass. Merging.

**Comment by moorepants** [comment]
fyi @ijxlyons

**Comment by moorepants** [comment]
fyi @ixjlyons

**Comment by moorepants** [comment]
@asmeurer What is needed to get this into the release branch?

**Comment by moorepants** [comment]
I probably should have changed the pull branch to your release branch, huh?

**Comment by asmeurer** [comment]
Can you PR the same branch to 1.1?

**Comment by asmeurer** [comment]
If you don't rebase it or anything it will be the same commit, so we can merge it in both and it won't be an issue.

**Comment by moorepants** [comment]
Ok I'll open a new PR from this branch onto the release branch.

**Pull request merged**